toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -290,7 +325,7 @@ func (r *RpmRepoCloner) Clone(cloneDeps bool, packagesToClone ...*pkgjson.Packag
 		err = r.chroot.Run(func() (err error) {
 			var chrootErr error
 			// Consider the built RPMs first, then the already cached (e.g. tooolchain), and finally all remote packages.
-			repoOrderList := []string{builtRepoID, cacheRepoID, allRepoIDs}
+			repoOrderList := []string{builtRepoID, fetcherRepoID, cacheRepoID, allRepoIDs}
```

Comment:
The source of this mess with using different folders for different build type is an issue with how our build pipelines are set-up and in general is a bug/misconfiguration. We need your change here and below as a workaround until that bug gets fixed. To make it easier to clean-up the code afterwards, though, could I ask you to make a small change, so we keep the workarounds closer to each other?

My ask: instead of adding `fetcherRepoID` to the `repoOrderList`, could we instead create an `effectiveCacheRepoID` variable and set it to either `cacheRepoID` or `fetcherRepoID` through that `if !buildpipeline.IsRegularBuild()` few lines above?

I'd have a similar ask for the changes in `WhatProvides`.

Reply:
Sounds reasonable, will do!